app/assets/javascripts/admin/property_toggle_ajax_handler.js

```diff
@@ -0,0 +1,88 @@
+function PropertyToggleAjaxHandler(options) {
+  this.actionsContainer = options.actionsContainer;
+  this.publishLinkAttribute = options.publishLinkAttribute;
+  this.unpublishLinkAttribute = options.unpublishLinkAttribute;
+  this.csrfToken = options.csrfToken;
+}
+
+PropertyToggleAjaxHandler.prototype.publishAjaxRequest = function(event) {
+  event.preventDefault();
+  _this = this;
+  $.ajax({
+    url: $(event.target).attr('data-publish-url'),
+    type: 'PUT',
+    data:  JSON.stringify({ id: $(event.target).attr('data-id') }),
+    contentType: 'application/json',
+    dataType: 'json',
+    beforeSend: function (xhr) {
+      xhr.setRequestHeader('X-CSRF-Token', _this.csrfToken)
+    },
+    error: function(XMLHttpRequest, errorTextStatus, error) {
+      console.log("Failed: " + errorTextStatus + " ;" + error);
+    },
+    success: function(dealJson) {
+      if(dealJson.status === 'error') {
+        _this.errorDiv.text(dealJson.error_message).removeClass('hidden');
```

Comment:
why do we need to handle this in a separate error status? simply use error callback for this